src/staticdata.c

```diff
@@ -2726,14 +2898,14 @@ static void jl_save_system_image_to_stream(ios_t *f, jl_array_t *mod_array,
         if (worklist == NULL) {
             for (i = 0; tags[i] != NULL; i++) {
                 jl_value_t *tag = *tags[i];
-                jl_queue_for_serialization(&s, tag);
+                jl_queue_for_serialization(&s, tag, /* retainer */ NULL);
```

Comment:
So the idea is that you enable debugging from individual sites? I was expecting that one would always pass `retainer` but that it might be NULL if you're not debugging. It might be important to explain this wherever `DEBUG_RETAINERS` gets documented.